evm_loader/cli/src/account_storage.rs

```diff
@@ -526,25 +587,31 @@ impl<'a> AccountStorage for EmulatorAccountStorage<'a> {
 
     fn code_size(&self, address: &Address) -> usize {
         info!("code_size {address}");
+
         self.ethereum_account_map_or(address, 0, |a| a.code_size as usize)
     }
 
     fn code_hash(&self, address: &Address) -> [u8; 32] {
-        use solana_sdk::keccak::hash;
-
         info!("code_hash {address}");
 
-        self.ethereum_contract_map_or(address, <[u8; 32]>::default(), |c| {
-            hash(&c.code()).to_bytes()
-        })
+        solana_sdk::keccak::hash(&self.code(address)).to_bytes()
     }
 
     fn code(&self, address: &Address) -> evm_loader::evm::Buffer {
         use evm_loader::evm::Buffer;
 
         info!("code {address}");
 
-        self.ethereum_contract_map_or(address, Buffer::empty(), |c| Buffer::new(&c.code()))
+        self.ethereum_contract_map_or(address, Buffer::empty(), |c| {
```

Comment:
ndb: maybe try to remove hadooken with something like this?

```rust
|c| {
    self.state_overrides
        .and_then(|account_overrides| account_overrides.get(address)?.code.as_ref())
        .map(|code| Buffer::new(&code.0))
        .unwrap_or_else(|| Buffer::new(&c.code()))
}
```

Reply:
@raventid thank you, fixed!